crates/hir/src/code_model.rs

```diff
@@ -122,6 +124,33 @@ impl Crate {
     pub fn all(db: &dyn HirDatabase) -> Vec<Crate> {
         db.crate_graph().iter().map(|id| Crate { id }).collect()
     }
+
+    /// Try to get the root URL of the documentation of a crate.
+    pub fn get_doc_url(self: &Crate, db: &dyn HirDatabase) -> Option<String> {
+        // Look for #![doc(html_root_url = "...")]
+        let attrs = db.attrs(AttrDef::from(self.root_module(db)).into());
+        let doc_attr_q = attrs.by_key("doc");
+
+        let doc_url = if doc_attr_q.exists() {
```

Comment:
Let's use early return here.

Added this to style-guide: https://github.com/rust-analyzer/rust-analyzer/pull/5852